### ISSUE TITLE
[Bug] CI failing - unit test failure in GitHub Actions

### ISSUE THREAD

**Issue by mikehardy** [ankidroid/Anki-Android#9173]
Unit tests are currently failing in GitHub CI, but they work locally.

Something appears to be wrong with the robolectric shadow view

It's possible local environments have some valid caching going on, while GitHub CI either has an invalid cache or is downloading something fresh (that is not being downloaded locally) causing it to fail only in CI.

Attempted to invalidate gradle cache to fix, it did not work - https://github.com/ankidroid/Anki-Android/commit/33473d9408e0fccc7ba6d6277ce691d0a53c4cc3

It is possible the robolectric downloader is doing something wrong: https://github.com/ankidroid/Anki-Android/blob/master/AnkiDroid/robolectricDownloader.gradle

I am investigating

```

com.ichi2.anki.ActivityStartupUnderBackupTest > activityHandlesRestoreBackup[Reviewer] FAILED
    java.lang.ExceptionInInitializerError
        at com.ichi2.anki.cardviewer.GestureProcessor.<init>(GestureProcessor.java:42)
        at com.ichi2.anki.AbstractFlashcardViewer.<init>(AbstractFlashcardViewer.java:350)
        at com.ichi2.anki.Reviewer.<init>(Reviewer.java:100)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at org.robolectric.util.ReflectionHelpers.callConstructor(ReflectionHelpers.java:405)
        at org.robolectric.Robolectric.buildActivity(Robolectric.java:89)
        at com.ichi2.testutils.ActivityList$ActivityLaunchParam.build(ActivityList.java:139)
        at com.ichi2.anki.ActivityStartupUnderBackupTest.activityHandlesRestoreBackup(ActivityStartupUnderBackupTest.java:78)

        Caused by:
        java.lang.NullPointerException
            at org.robolectric.shadows.ShadowViewConfiguration.setup(ShadowViewConfiguration.java:75)
            at org.robolectric.shadows.ShadowViewConfiguration.get(ShadowViewConfiguration.java:93)
            at android.view.ViewConfiguration.get(ViewConfiguration.java)
            at com.ichi2.anki.reviewer.GestureMapper.<clinit>(GestureMapper.java:38)
            at com.ichi2.anki.cardviewer.GestureProcessor.<init>(GestureProcessor.java:42)
            at com.ichi2.anki.AbstractFlashcardViewer.<init>(AbstractFlashcardViewer.java:350)
            at com.ichi2.anki.Reviewer.<init>(Reviewer.java:100)
            at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
            at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
            at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
            ... 4 more
```

**Comment by mikehardy** [comment]
>             at org.robolectric.shadows.ShadowViewConfiguration.setup(ShadowViewConfiguration.java:75)

is 

```java
    DisplayMetrics metrics = context.getResources().getDisplayMetrics();
```

Trying to figure out why the context is busted currently. This appears to be a headless mode issue, been a while since I've seen one of those!

**Comment by mikehardy** [comment]
1. Attempted reproduction by simply `ssh localhost` then re-running `./gradlew jacocoUnitTestReport: success, though `echo $DISPLAY` did indicate there was no X available and `xterm` failed. I'm on Ubuntu 21. I thought that would reproduce it, I was surprised it worked.
2. Attempted reproduction by installing docker, pulling Ubuntu 20LTS (matches Github Actions Ubuntu runner), installing adoptopenjdk11-hotspot VM (following their instructions), installing git, `git clone https://github.com/AnkiDroid/Anki-Android.git, `cd Anki-Android`, installing android cmdline-tools, svdmanager accepting licenses, `./gradlew jacocoTestReport`. It also succeeds! Very unexpected

So I don't have a reproduction yet. Still investigating.

**Comment by mikehardy** [comment]
Other priorities at the moment, this will sit for a bit and I commit to manually verifying PRs until resolved.
It is just about highest priority for me w.r.t. AnkiDroid just have other stuff to turn attention to for a while

**Comment by mikehardy** [comment]
I am less surprised now by these unexpected successful reproductions but this also worked locally, when I thought it might faill

`/gradlew -Djava.awt.headless=true clean jacocoUnitTestReport --rerun-tasks`

**Comment by mikehardy** [comment]
As a potential avenue of fix (though I don't like it since it is working around a tool issue instead of fixing the tool issue...)

https://github.com/ankidroid/Anki-Android/commit/42ab45ce31c43f8b9d05129303d42c1fb506b488#diff-a58e9d82dec419c29f3765e76012c22af32ff1a9ab13c797276c1300a1ff7be0R44-R49

That is the static initializer that is failing, and if the fetch on ViewConfiguration were made non-static (as it was before) it means that we would be fetching ViewConfiguration every time during GestureMapper.init, but that is only called during GestureProcessor.init which itself is only called on AbstractFlashcardViewer.restorePreferences - so it would not be that expensive and in fact would match previous before.

In other words, we would lose a slight optimization in order to allow robolectric to initialize itself correctly

Second possible avenue is to forcibly initialize that chunk of robolectric prior to instantiating any GestureMappers, and that's what I'm more keen on doing.

Ongoing investigation
